bulk_site_creator/views.py

```diff
@@ -0,0 +1,322 @@
+SDK_CONTEXT = RequestContext(**SDK_SETTINGS)
+
+dynamodb = boto3.resource("dynamodb")
+table_name = settings.BULK_COURSE_CREATION.get("site_creator_dynamo_table_name")
+
+# TODO improve logging for CRUD actions
+# TODO add documentation to each view
+
+@login_required
+@lti_role_required(const.ADMINISTRATOR)
+@lti_permission_required(settings.PERMISSION_BULK_SITE_CREATOR)
+@require_http_methods(["GET", "POST"])
+def index(request):
+    """
+    This function renders the Bulk Site Creator index page
+    consisting of bulk creation jobs (if any).
+    """
+    table = dynamodb.Table(table_name)
+    sis_account_id = request.LTI["custom_canvas_account_sis_id"]
+    school_id = sis_account_id.split(":")[1]
+    school_key = f'SCHOOL#{school_id.upper()}'
+    query_params = {
+        'KeyConditionExpression': Key('pk').eq(school_key),
+        'ScanIndexForward': False,
+    }
+    logger.debug(f'Retrieving jobs for school {school_key}.')
+    jobs_for_school = table.query(**query_params)['Items']
+
+    # Update string timestamp to datetime.
+    [item.update(created_at=parse_datetime(item['created_at']))
+     for item in jobs_for_school]
+
+    context = {
+        'jobs_for_school': jobs_for_school
+    }
+    logger.debug(f'Retrieved jobs for school {school_key}.', extra=context)
+    return render(request, "bulk_site_creator/index.html", context=context)
+
+
+@login_required
+@lti_role_required(const.ADMINISTRATOR)
+@lti_permission_required(settings.PERMISSION_BULK_SITE_CREATOR)
+@require_http_methods(["GET"])
+def job_detail(request: HttpRequest, job_id: str) -> HttpResponse:
+    """
+    This function renders the Bulk Site Creator job detail page 
+    with details about a job and the tasks for that job.
+    """
+    table = dynamodb.Table(table_name)
+    sis_account_id = request.LTI["custom_canvas_account_sis_id"]
+    school_id = sis_account_id.split(":")[1]
+    school_key = f'SCHOOL#{school_id.upper()}'
+    job_query_params = {
+        'KeyConditionExpression': Key('pk').eq(school_key) & Key('sk').eq(job_id),
+        'ScanIndexForward': False,
+    }
+    logger.debug(f'Retrieving job details for job {job_id}.')
+    job = table.query(**job_query_params)['Items'][0]
+
+    # Update string timestamp to datetime.
+    job.update(created_at=parse_datetime(job['created_at']))
+    job.update(updated_at=parse_datetime(job['updated_at']))
+
+    tasks_query_params = {
+        'KeyConditionExpression': Key('pk').eq(job_id),
+        'ScanIndexForward': False,
+    }
+    tasks = table.query(**tasks_query_params)['Items']
+
+    context = {
+        'job': job,
+        'tasks': tasks
+    }
+    logger.debug(f'Retrieved job details for job {job_id}.', extra=context)
+    return render(request, "bulk_site_creator/job_detail.html", context=context)
+
+
+@login_required
+@lti_role_required(const.ADMINISTRATOR)
+@lti_permission_required(settings.PERMISSION_BULK_SITE_CREATOR)
+@require_http_methods(["GET", "POST"])
+def new_job(request):
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This function calls 17 other functions. By comparison, 98% of the functions in the CodeGuru reference dataset use fewer. This indicates the function is highly coupled with other functions. A function that is highly coupled with other functions is difficult to understand and its behavior might change unexpectedly when one of its referenced functions is updated. High coupling could also increase the integration test complexity, maintenance cost and technical debt.

We recommend that you simplify this function or break it into multiple functions. For example, consider extracting the code block on lines 176-196 into a separate function.